src/core/utils/format-number.ts

```diff
@@ -47,3 +47,6 @@ export const formatNumber = (amount: number | BigNumber, options: INumberFormatO
         ? formattedNumber + ' ' + options.currency
         : formattedNumber;
 };
+
+const numericRegex = new RegExp(/^[+-]?\d+(\.\d+)?$/);
```

Comment:
is this what we want ? you should understand regex expressions you use :P
try https://regexr.com/

Reply:
try to understand how this regex rule works, after that i will tell you how to do it without regex :)